app/src/main/java/com/example/kienphung/lesson4/MainActivity.java

```diff
@@ -1,13 +1,85 @@
 package com.example.kienphung.lesson4;
 
-import android.support.v7.app.AppCompatActivity;
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.AsyncTask;
 import android.os.Bundle;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.ArrayList;
 
 public class MainActivity extends AppCompatActivity {
+    private static final String[] MODIFIER = {".jpg", ".jpeg", ".png"};
+    private static final int READ_EXTERNAL_STORAGE = 1;
+    private static final String[] PATH = {"/storage/emulated/0/Download/"};
+    private ArrayList<String> mPathFolder;
+    private ImageAdapter imageAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        checkPermission();
+        initView();
+        new loadView().execute(PATH);
+    }
+
+    private void checkPermission() {
+        if (ContextCompat.checkSelfPermission(this,
+                Manifest.permission.READ_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},
+                    READ_EXTERNAL_STORAGE);
+        }
+    }
+
+    private void initView() {
+        RecyclerView recyclerViewImages = findViewById(R.id.recycler_view);
+        mPathFolder = new ArrayList<>();
+        recyclerViewImages.setLayoutManager(new GridLayoutManager(this, 2));
+        imageAdapter = new ImageAdapter(mPathFolder);
+        recyclerViewImages.setAdapter(imageAdapter);
+    }
+
+    private class loadView extends AsyncTask<String, String, Void> {
```

Comment:
chữ cái đầu class phải in hoa
```suggestion
    private class GetImagesAsyncTask extends AsyncTask<String, String, Void> {
```